dashboard/test/integration/omniauth/facebook_test.rb

```diff
@@ -15,6 +15,33 @@ class FacebookTest < ActionDispatch::IntegrationTest
       SignUpTracking.stubs(:split_test_percentage).returns(0)
     end
 
+    test "student sign up for newest sign up flow" do
+      auth_hash = mock_oauth
+
+      post "/users/auth/facebook"
+      get '/users/auth/facebook/callback', params: {finish_url: '/users/new_sign_up/finish_student_account'}
```

Comment:
The utils file has some [sign_in_through](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/integration/omniauth/utils.rb#L40-L44) helpers that do the post to a provider and assert the redirect. Might dry this up a bit.

Reply:
I'm going to do some clean up in a followup and rework some of these tests to make sure they're truly testing the new flow. Merging this as is when staging reopens so I can start from this point!